examples/cms-starter/src/data.ts

```diff
@@ -0,0 +1,293 @@
+
+// Plugin keys
+export const PLUGIN_PREFIX = "cms_starter"
+export const LOCAL_STORAGE_LAST_LAUNCH_KEY = `${PLUGIN_PREFIX}.lastLaunched`
+
+export const PLUGIN_COLLECTION_SYNC_REFERENCE_KEY = `collectionSyncReference`
+export const PLUGIN_COLLECTION_SYNC_SLUG_KEY = `collectionSyncSlug`
+
+export type DataSourceFieldType =
+    | "string"
+    | "date"
+    | "image"
+    | "reference"
+    | "richText"
+    | "number"
+    | "boolean"
+    | "enum"
+    | "color"
+
+/**
+ * Mapping of data source field types to managed collection field types.
+ */
+export const FIELD_MAPPING: Record<DataSourceFieldType, SupportedCollectionFieldTypeWithoutReference[]> = {
+    string: [],
+    date: ["date"],
+    image: ["image", "file", "link"],
+    // Special case for reference fields - we need to map the reference field to the collection it references
+    reference: [],
+    richText: ["formattedText"],
+    number: ["number"],
+    boolean: ["boolean"],
+    enum: ["enum"],
+    color: ["color"],
+}
+
+export type DataSourceField =
+    | {
+          type: Exclude<DataSourceFieldType, "reference" | "enum">
+      }
+    | {
+          type: "reference"
+          reference: string
+          multiple: boolean
+      }
+    | {
+          type: "enum"
+          options: string[]
+      }
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export type SupportedCollectionFieldType = ManagedCollectionField["type"]
+export type SupportedCollectionFieldTypeWithoutReference = Exclude<
+    SupportedCollectionFieldType,
+    "collectionReference" | "multiCollectionReference"
+>
+
+/**
+ * Reference fields are special fields that reference other collections.
+ */
+export interface ReferenceField {
+    type: "collectionReference" | "multiCollectionReference"
+    source: string
+    destination: string | null
+}
+
+/**
+ * Field configuration for a managed collection field.
+ */
+export interface FieldConfig {
+    source: {
+        name: string
+        type: DataSourceFieldType
+        ignored: boolean
+    }
+    field: ManagedCollectionField | null
+    reference: ReferenceField | null
+}
+
+interface CollectionDetails {
+    id: string
+    name: string
+}
+export const COLLECTIONS_SYNC_MAP: Map<string, CollectionDetails[]> = new Map()
+
+export const allExistingCollections = await framer.getCollections()
+for (const collection of allExistingCollections) {
+    const reference = await collection.getPluginData(PLUGIN_COLLECTION_SYNC_REFERENCE_KEY)
+    if (reference) {
+        const collectionReferences = COLLECTIONS_SYNC_MAP.get(reference) ?? []
+        COLLECTIONS_SYNC_MAP.set(reference, [
+            ...collectionReferences,
+            {
+                id: collection.id,
+                name: collection.name,
+            },
+        ])
+    }
+}
+
+export function computeFieldConfig(existingFields: ManagedCollectionField[], dataSource: DataSource) {
```

Comment:
Can we avoid the need for the complexity of this function? Can we just map the type of the source data more directly to our internal types? This is quite confusing for an example given all the custom types it requires.